src/ray/core_worker/core_worker.cc

```diff
@@ -378,6 +378,17 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
                           : std::shared_ptr<LeasePolicyInterface>(
                                 std::make_shared<LocalLeasePolicy>(rpc_address_));
 
+  std::function<uint64_t()> get_max_pending_lease_requests_per_scheduling_category =
+      []() {
+        return RayConfig::instance().max_pending_lease_requests_per_scheduling_category();
+      };
+  if (RayConfig::instance().max_pending_lease_requests_per_scheduling_category() == -1) {
+    // Set the number of concurrent lease requests to the current number of
+    // nodes in the cluster.
+    get_max_pending_lease_requests_per_scheduling_category = [this]() {
+      return gcs_client_->Nodes().GetAll().size();
```

Comment:
```
const absl::flat_hash_map<NodeID, GcsNodeInfo> &NodeInfoAccessor::GetAll() const {
  return node_cache_;
}
```
The nodes returned by `GetAll()` contains dead nodes, which should be filtered out.